Sources/FoundationEssentials/ProcessInfo/ProcessInfo.swift

```diff
@@ -182,17 +213,117 @@ extension _ProcessInfo {
             return $0!
         }
     }
+    
+#if os(Windows)
+    internal var _rawOperatingSystemVersionInfo: RTL_OSVERSIONINFOEXW? {
+        guard let ntdll = ("ntdll.dll".withCString(encodedAs: UTF16.self) {
+            LoadLibraryExW($0, nil, DWORD(LOAD_LIBRARY_SEARCH_SYSTEM32))
+        }) else {
+            return nil
+        }
+        defer { FreeLibrary(ntdll) }
+        typealias RTLGetVersionTy = @convention(c) (UnsafeMutablePointer<RTL_OSVERSIONINFOEXW>) -> NTSTATUS
+        guard let pfnRTLGetVersion = unsafeBitCast(GetProcAddress(ntdll, "RtlGetVersion"), to: Optional<RTLGetVersionTy>.self) else {
+            return nil
+        }
+        var osVersionInfo = RTL_OSVERSIONINFOEXW()
+        osVersionInfo.dwOSVersionInfoSize = DWORD(MemoryLayout<RTL_OSVERSIONINFOEXW>.size)
+        guard pfnRTLGetVersion(&osVersionInfo) == 0 else {
+            return nil
+        }
+        return osVersionInfo
+    }
+#endif
 
     var operatingSystemVersionString: String {
         // TODO: Check for `/etc/os-release` for Linux once DataIO is ready
         // https://github.com/apple/swift-foundation/issues/221
-        #if os(macOS)
+#if os(macOS)
         var versionString = "macOS"
-        #elseif os(Linux)
+#elseif os(Linux)
+        if let osReleaseContents = try? Data(contentsOf: "/etc/os-release") {
+            let strContents = String(decoding: osReleaseContents, as: UTF8.self)
+            if let name = strContents.split(separator: "\n").first(where: { $0.hasPrefix("PRETTY_NAME=") }) {
+                // This is extremely simplistic but manages to work for all known cases.
+                return String(name.dropFirst("PRETTY_NAME=".count)._trimmingCharacters(while: { $0 == "\"" }))
+            }
+        }
+        
+        // Okay, we can't get a distro name, so try for generic info.
         var versionString = "Linux"
-        #else
-        var versionString = ""
-        #endif
+#elseif os(Windows)
+        var versionString = "Windows"
+        
+        guard let osVersionInfo = self._rawOperatingSystemVersionInfo else {
+            return versionString
+        }
+
+        // Windows has no canonical way to turn the fairly complex `RTL_OSVERSIONINFOW` version info into a string. We
+        // do our best here to construct something consistent. Unfortunately, to provide a useful result, this requires
+        // hardcoding several of the somewhat ambiguous values in the table provided here:
+        //  https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/ns-wdm-_osversioninfoexw#remarks
+        switch (osVersionInfo.dwMajorVersion, osVersionInfo.dwMinorVersion) {
```

Comment:
Might be nice to use the switch expressions that were recently introduced to convert the tuple to a string and append once rather than in each case.